Workflows/Environment Synchronization.md

```diff
@@ -0,0 +1,126 @@
+A synchronization of environments, or more precisely, a synchronization of the data used in different environments,
+is usually necessary when you need to work in an environment that is as production-like as possible, but you don't have
+access to production itself, for example:
+* when you need to debug a more complex issue occuring in production
+* when you need data that is as close to production as possible (both when it comes to type and amount)
+for development and/or testing purposes
+
+Access to production is normally limited in order to protect the private information of users, among other things.
+The same should be ensured once this data is copied to another environment. In order to limit the number
+of developers who have access to raw production data, it would be best if that data was anonymized by someone
+from the DevOps team before it is given to the developer. To achieve this, the Rails team has agreed with
+the DevOps team that the synchronization process should look like this:
+1. a completely new environment is set up
+2. the production database is copied to this new environment
+3. anonymization is run on the copied production database
+4. the anonymized database is dumped
+5. the anonymized database dump is copied to a destination of choice (e.g. S3 bucket)
+6. the anonymized database dump is downloaded to the destination environment (staging, development, etc.)
+7. a dump of the local database is generated as a backup
+8. the local database schema is reset
+9. the anonymized database dump is imported/restored to the local database
+
+Keep in mind that all the steps up until and including step #5 require the involvement of the DevOps team.
+The remaining steps can be automated using the copy_bot gem, which is explained later in this chapter.
+
+In addition to this, for the devops team to be able to anonymize the production database, you first need to provide
+them with a way to do that. One such solution is described in the following section.
+
+### Data anonymization
+Data anonymization is a form of information sanitization intended to protect private information contained in
+the database. It can be performed when it is necessary to work with production data during testing and/or debugging of
+an issue, but also to fulfill GDPR requirements. In the first case, personal information is anonymized only when it is
+needed (e.g. after copying the production database) and only in non-production environments. However, in the case of
+GDPR compliance, it may need to be done in regular intervals in the production environment.
+
+The required final result may also be different depending on the use case.
+It may be enough to replace the sensitive data with random strings, but sometimes this is not a good solution if it
+prevents the app from functioning properly. In the latter case it is more appropriate to replace real data with
+fake data: fake names to replace the names of real persons, fake phone numbers
```

Comment:
Synthetic could be used here instead of fake, what do you think @cukoreva ?

Reply:
Faker is included here just as an example, it's not part of the gem, and the developer implementing the anonymization in their project can choose whatever they want (they don't even have to go with a gem, but can implement some logic of their own).

Reply:
I actually meant the "fake" adjective by itself, not the Faker gem references. Fake has a bit of negative connotation in my head

Reply:
I get it now :) I'll update the sentence.